appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/ResourceValidator.java

```diff
@@ -0,0 +1,696 @@
+            }
+
+            for (Object next : bd.getMessageDestinations()) {
+                accept((MessageDestinationDescriptor) next, nameEnvironment);
+            }
+
+            for (Object next : nameEnvironment.getEnvironmentProperties()) {
+                accept((EnvironmentProperty) next, nameEnvironment);
+            }
+
+            for (Object next : nameEnvironment.getEjbReferenceDescriptors()) {
+                accept((EjbReferenceDescriptor) next, nameEnvironment);
+            }
+
+            for (PersistenceUnitsDescriptor pus : bd.getExtensionsDescriptors(PersistenceUnitsDescriptor.class)) {
+                for (PersistenceUnitDescriptor pu : pus.getPersistenceUnitDescriptors()) {
+                    accept(pu, nameEnvironment);
+                }
+            }
+
+            for (ManagedBeanDescriptor mbd: bd.getManagedBeans()) {
+                accept(mbd, nameEnvironment);
+            }
+        }
+    }
+
+    /**
+     * Validate resources stored in ResourceRefDescriptor.
+     */
+    private void accept(ResourceReferenceDescriptor resRef, JndiNameEnvironment env) {
+        String jndiName = resRef.getJndiName();
+
+        if (resRef.isWebServiceContext())
+            return;
+
+        if (resRef.isURLResource()) {
+            if (jndiName != null && !(jndiName.startsWith(ResourceConstants.JAVA_SCOPE_PREFIX))) {
+                try {
+                    // for jndi-name like "http://localhost:8080/index.html"
+                    new java.net.URL(jndiName);
+                    return;
+                } catch (MalformedURLException e) {
+                    // If jndi-name is not an actual url, we might want to lookup the name
+                }
+            }
+        }
+        validateJNDIRefs(resRef.getJndiName(), env);
+    }
+
+    /**
+     * Validate resources stored in ResourceEnvRefDescriptor.
+     * Managed Bean references are validated here.
+     */
+    private void accept(ResourceEnvReferenceDescriptor resourceEnvRef, JndiNameEnvironment env) {
+        String jndiName = resourceEnvRef.getJndiName();
+        if (resourceEnvRef.isEJBContext() || resourceEnvRef.isValidator() || resourceEnvRef.isValidatorFactory() || resourceEnvRef.isCDIBeanManager())
+            return;
+
+        // Validate Managed Bean references now
+        String newName = convertModuleOrAppJNDIName(jndiName, env);
+        if (!myNamespace.find(newName, env)) {
+            // Every type of resource taken care of. Validate any custom resources now.
+            validateJNDIRefs(jndiName, env);
+        }
+    }
+
+    /**
+     * Convert JNDI names beginning with java:module and java:app to their corresponding java:global names.
+     *
+     * @return the converted name with java:global JNDI prefix.
+     */
+    private String convertModuleOrAppJNDIName(String jndiName, JndiNameEnvironment env) {
+        BundleDescriptor bd = null;
```

Comment:
It is possible. But I have written code for the only two cases we are concerned about.